ql/src/java/org/apache/hadoop/hive/ql/session/SessionState.java

```diff
@@ -1852,6 +1863,9 @@ public void close() throws IOException {
       closeSparkSession();
       registry.closeCUDFLoaders();
       dropSessionPaths(sessionConf);
+      if (pathCleaner != null) {
+        pathCleaner.shutdown();
```

Comment:
dropSessionPaths() would have created the thread and would have started deleting. As shutdown() is called immediately, it would stop further processing and quit without purging other entries. (i.e This would prevent the thread from processing > 1024 entries in PathCleaner. It would be good to wait until all files are purged before quitting the thread.

Reply:
I dont see how this happens. Everything that needs to be deleted is guaranteed to be in PathCleaner's deleteActions list by the time shutdown is called. PathCleaner goes over everything in the list before exiting. See: https://github.com/apache/hive/pull/1577/files#diff-80bcef5e921dc6ac67a42ad691b771943f0ffa608454dae389c76b29139c61f9R65 

I didn't get your point with 1024 entries either. That is just the initial capacity of intermediate ArrayList. That is not an upper bound.